mayastor/src/subsys/mbus/mbus_nats.rs

```diff
@@ -0,0 +1,100 @@
+//! NATS implementation of the `MessageBus` connecting mayastor to the control
+//! plane components.
+
+use super::MessageBus;
+use async_trait::async_trait;
+use nats::asynk::Connection;
+use once_cell::sync::OnceCell;
+use smol::io;
+
+pub(super) static NATS_MSG_BUS: OnceCell<NatsMessageBus> = OnceCell::new();
+pub(super) fn message_bus_init(server: String) {
+    NATS_MSG_BUS.get_or_init(|| {
+        // Waits for the message bus to become ready
+        tokio::runtime::Runtime::new()
+            .unwrap()
+            .block_on(async { NatsMessageBus::new(&server).await })
```

Comment:
So here we block the reactor until mayastor connects to the message bus, right? That's probably ok, just asking.

Reply:
Yes that's right, this was to avoid having the message bus deal with it not being connected the first time. After the first connect then the library handles that for us.